gp/primitives.py

```diff
@@ -26,33 +23,24 @@ def __init__(self, primitives: List[Type[Node]]) -> None:
             if not issubclass(primitives[i], Node):
                 raise TypeError(f"expected subclass of Node but received {primitives[i].__name__}")
 
-        self._primitives = {}
-        for i in range(len(primitives)):
-            self._primitives[i] = primitives[i]
-
-        # hide primitives dict behind MappingProxyType to make sure it
-        # is not changed after construction
-        # unfortunately not supported by pickle, necessary for
-        # multiprocessing; another way to implement this?
-        # self._primitives = types.MappingProxyType(self._primitives)
+        self._primitives = list(primitives)
```

Comment:
If `primitives` is already a list, why calling `list()` here?

Reply:
to create a shallow copy of a list

Reply:
Got it.